cocos/particle/renderer/particle-system-renderer-cpu.ts

```diff
@@ -716,14 +718,16 @@ export default class ParticleSystemRendererCPU extends ParticleSystemRendererBas
                 this._trailDefines[CC_USE_WORLD_SPACE] = false;
             }
             let mat = ps.getMaterialInstance(1);
-            if (mat === null && this._defaultTrailMat === null) {
-                _matInsInfo.parent = builtinResMgr.get<Material>('default-trail-material');
-                _matInsInfo.owner = this._particleSystem;
-                _matInsInfo.subModelIdx = 1;
-                this._defaultTrailMat = new MaterialInstance(_matInsInfo);
-                _matInsInfo.parent = null!;
-                _matInsInfo.owner = null!;
-                _matInsInfo.subModelIdx = 0;
+            if (Editor) {
+                if (mat === null && this._defaultTrailMat === null) {
```

Comment:
I don't think it can be changed that way, the user may not have any materials in the old project, the previous version worked, but now it doesn't work.